Models/ImageEncoder.py

```diff
@@ -25,19 +25,33 @@ def __init__(self, choice, dropout=False, fine_tune=False):
       # GoogLeNet
       self.model_name = "GoogLeNet_Inception_v3"
       model_ft = models.inception_v3(pretrained=True)
-     
+      self.model = model_ft.to(self.device)  
+      # remove the dropout + linear classifier layer
+      layers = list(self.model.to(self.device).children())[:-2]
+      self.model = nn.Sequential(*layers)
+
     elif choice == "AlexNet":
       self.model_name = "AlexNet"
       model_ft = models.alexnet(pretrained=True)
+      self.model = model_ft.to(self.device)  
+      # remove classifier layer
+      net = list(model_ft.children())[-1]
+      net_layers = list(net.children())[:-2]
+      net = nn.Sequential(*net_layers)
+      model_layers = list(model_ft.children())[:-1] + [net]
```

Comment:
What is this doing exactly? I didn't understand.
It seems like net is the last FC layer, net layers if the model minus the last two layers, model_layers is just the full model again?